http-server/src/main/clojure/xtdb/server.clj

```diff
@@ -104,6 +106,33 @@
             (let [writer (transit/writer out :json opts)]
               (transit/write writer res))))))))
 
+(defn- ->jsonl-resultset-encoder [_opts]
+  (reify
+    mf/EncodeToBytes
+    ;; we're required to be a sub-type of ETB but don't need to implement its fn.
+
+    mf/EncodeToOutputStream
+    (encode-to-output-stream [_ res _]
+      (fn [^OutputStream out]
+        (if-not (ex-data res)
+          (with-open [^IResultSet res res
+                      writer (io/writer out)]
+            (try
+              (doseq [el (iterator-seq res)]
+                (json/write el writer)
```

Comment:
If we're expecting users to wrap input types in `{"@value": ..., "type": "..."}` (or similar), we should do the same on output so that the values round-trip correctly.

Reply:
I agree. Can we do this via a postwalk before writing the value for now?

Reply:
Jackson it is.